### PR TITLE
value::asLong will convert from string to long

### PR DESCRIPTION
**Pull request by mrkline**
This is less surprising behavior when calling asLong() on an arg expected to be an integer value.

See #27.

### CODE REVIEW

**Comment by mrkline** [comment]
If desired, something similar could be done for `asBool`, mapping "true", "t", "yes", "y" etc. to `true` and "false", "f", "no", "n", etc. to `false`.

**Comment by Jaxan** [comment]
In my fork I have exactly the same change. This works very well and is super easy, so please merge this!

btw, there is some discussion on this issue here as well: https://github.com/docopt/docopt.cpp/issues/8

**Comment by jaredgrubb** [comment]
My initial reaction to this is that this causes ambiguity in that you can no longer tell between "-vvv" (current 'asLong' behavior) and "-x3" (current string arg behavior) because "asLong" will now give integer 3 in both cases ... but I really cant justify that at all. So, let's do it!

File: docopt_value.h [hunk 1]
```diff
@@ -257,6 +257,13 @@ namespace docopt {
 	inline
 	long value::asLong() const
 	{
+		// Attempt to convert a string to a long
+		if (kind == Kind::String) {
+			// Doesn't guard against trailing characters,
+			// but doing so (if desired) would be trivial by checking pos.
+			std::size_t pos;
+			return stol(variant.strValue, &pos); // Throws if it can't convert
```
Comment (jaredgrubb) [inline docopt_value.h]: Should we throw here if "pos != strValue.size()"? I'm not sure I like that "--foo 42bar" would give "42" and silently discard the junk at the end.

Comment (mrkline) [inline docopt_value.h]: Certainly. Pushed.

**Comment by jaredgrubb** [comment]
Looks good... let's do it!

**Pull request merged**